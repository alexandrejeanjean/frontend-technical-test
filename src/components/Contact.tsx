import { Conversation } from "../types/conversation"

type Props = {
  conversation: Conversation;
  handleConversationSelection: (conversation: Conversation) => void;
}

const Contact = ({ conversation, handleConversationSelection }: Props) => {

  return (
    <button onClick={() => handleConversationSelection(conversation)} className='inline-flex items-center w-full h-full px-3 py-4 border-b-2 border-gray-50 hover:bg-indigo-200 focus:bg-indigo-300 active:bg-indigo-100'>
      <div className='flex items-center justify-center w-12 h-12 mr-3 bg-indigo-700 rounded-full'>
        <p className="font-medium text-white">{conversation.senderNickname[0]}</p>
      </div>
      <p className='flex-1 font-medium text-left text-gray-700'>{conversation.senderNickname}</p>
      <span className='arrow right '></span>
    </button>
  )
}

export default Contact
