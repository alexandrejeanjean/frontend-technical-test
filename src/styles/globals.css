@tailwind base;
@tailwind components;
@tailwind utilities;

.arrow {
  @apply block p-1 border-t-0 border-b-2 border-l-0 border-r-2 border-gray-500 border-solid;
}

.right {
  transform: rotate(-45deg);
  -webkit-transform: rotate(-45deg);
  @apply border-indigo-600;
}

.main-container-height {
  height: calc(100vh - 132px);
}

.list-height {
  height: calc(100% - 50px);
}