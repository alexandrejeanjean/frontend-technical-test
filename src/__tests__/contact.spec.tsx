import { expect, jest } from '@jest/globals';
import { mount } from 'enzyme';
import Contact from "../components/Contact";

const mockConversation = {
    id: 0,
    lastMessageTimestamp: 1636306788437,
    recipientId: 0,
    recipientNickname: "Théo Dort",
    senderId: 0,
    senderNickname: "Alexandre Jeanjean"
};

const handleConversationSelection = jest.fn();

/** @test {Contact Component} */
describe('Contact Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = mount(<Contact conversation={mockConversation} handleConversationSelection={handleConversationSelection} />);
    });

    it('should render without crashing', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('should display the good value', () => {
        expect((wrapper).prop('conversation')).toEqual({
            id: 0,
            lastMessageTimestamp: 1636306788437,
            recipientId: 0,
            recipientNickname: "Théo Dort",
            senderId: 0,
            senderNickname: "Alexandre Jeanjean"
        });
    });


    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

});

