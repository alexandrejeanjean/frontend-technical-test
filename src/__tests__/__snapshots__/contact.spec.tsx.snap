// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component match snapshot 1`] = `ReactWrapper {}`;
